packages/flutter/lib/src/rendering/view.dart

```diff
@@ -201,9 +202,17 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     // Layer hit testing is done using device pixels, so we have to convert
```

Comment:
Why do we still need to have a separate hit testing pass for mouse events? Can that not be integrated with the regular hit testing now as well?

Reply:
Without mouse,

* During hover events, no regular hit testing is done.
* During down events, regular hit testing is done, so this is an extra pass, but down events happen at very low frequency that we might not really care.
* During move events, no regular hit testing is done (they use the result of down events).

Conclusion: The only "extra pass" is the ones at down events, which do not really matter.

Reply:
I am more concerned about the extra code complexity that we have to deal with because the paths are not unified.

Reply:
If the mouse device update is triggered by a pointer event, then the code path is indeed more complicated than necessary: 
```

GestureBinding.dispatchEvent -> PointerRouter.route -> MouseTracker._handleEvent -> hit test
```
Maybe we could've done the hit test at `GestureBinding.dispatchEvent` and pass it all the way to `MouseTracker._handleEvent`.

But the mouse device update can also be triggered after every frame, during the post-frame callbacks:
```
SchedulerBinding.handleDrawFrame -> MouseTracker._updateAllDevices -> hit test
```
In this case `MouseTracker` have to initiate a hit test for each device, without the participation of `GestureBinding` at all. 

I'm not saying that a much more unified structure can not be achieved at all, but for now I'd rather touch as little things as possible in order to see the performance impact of the current PR.